app/(app)/offence/index.tsx

```diff
@@ -49,39 +55,53 @@ const OffencePage = () => {
     [role?.actions.zone, offenceType, location, udrData],
   )
 
-  const onSubmit = async () => {
-    if (isSubmitting) {
-      return
-    }
-
-    setIsSubmitting(true)
-    setIsTouched(true)
-
-    if (!(offenceType && (isObjectiveResponsibility || resolutionType) && ecv) || isLocationError) {
-      setIsSubmitting(false)
-
-      return
-    }
-
-    const response = await clientApi.scanControllerGetOffenceList(
-      ecv,
-      undefined,
-      undefined,
-      location?.lat,
-      location?.long,
-    )
-
-    if (response.data.length > 0) {
-      openModal()
-      setIsSubmitting(false)
-
-      return
-    }
-
-    router.navigate('/offence/vehicle')
-    setIsSubmitting(false)
-    setIsTouched(false)
-  }
+  const { mutate, isPending, isError } = useMutation({
+    mutationFn: async () => {
+      if (
+        !(offenceType && (isObjectiveResponsibility || resolutionType) && ecv && scanData) ||
+        isLocationError
+      ) {
+        throw new Error('Missing required data')
+      }
+
+      let scanResponse
+
+      // Update scan if ECV was updated manually
+      if (scanData.ecv !== ecv) {
+        scanResponse = await clientApi.scanControllerCreateOrUpdateScanEcv({
+          ...scanData,
+          ecv,
+          ecvUpdatedManually: true,
+          udr: scanData.udr ?? undefined,
+          streetName: scanData.streetName ?? undefined,
+        })
+      }
+
+      const duplicityResponse = await clientApi.scanControllerGetOffenceList(
+        ecv,
+        undefined,
+        undefined,
+        location?.lat,
+        location?.long,
+      )
+
+      return { scanResponseData: scanResponse?.data, duplicityResponseData: duplicityResponse.data }
+    },
+    onSuccess: ({ duplicityResponseData, scanResponseData }) => {
+      if (scanResponseData) {
+        setOffenceState({
```

Comment:
Just an idea, in the future it would probably be better to use React Query for storing this data instead of propagating this to global store. There are really nice features for these kind of things:

https://tanstack.com/query/v4/docs/framework/react/guides/invalidations-from-mutations
https://tanstack.com/query/v4/docs/framework/react/guides/updates-from-mutation-responses
https://tanstack.com/query/v4/docs/framework/react/guides/optimistic-updates

Or maybe I'm misreading what it does.